### ISSUE TITLE
Refresh layer on notification for postgres layer freezes UI

### ISSUE THREAD

**Issue by nextstopsun** [qgis/QGIS#55707]
### What is the bug or the crash?

In the `Layer Properties` panel for a postgres layer when setting `Refresh layer on notification` checkbox and then hitting `Apply` button UI freezes QGIS forever. So only QGIS termination helps.

### Steps to reproduce the issue

1. Add layer from postgresql db
2. Open `Layer Properties` window
3. Go to `Rendering tab`
4. Check `Refresh layer on notification`
5. UI Freezes

### Versions

Tested on versions from 3.26 up to 3.34

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

Tried also on GeoPackage layer. The freeze happens only for postgres layers.

**Comment by troopa81** [comment]
Duplicate of #54260. It will be fixed once https://github.com/qgis/QGIS/pull/55062  is merged
